auth/auth.go

```diff
@@ -227,6 +233,11 @@ type CredentialsOptions struct {
 	QuotaProjectIDProvider CredentialsPropertyProvider
 	// UniverseDomainProvider resolves the universe domain with the credentials.
 	UniverseDomainProvider CredentialsPropertyProvider
+	// TrustBoundaryDataProvider is used to fetch trust boundary data.
+	// This data defines location restrictions for credential usage.
+	// Not all credential types utilize this provider; if it's nil, no trust boundary
+	// restrictions are applied via this mechanism.
+	TrustBoundaryDataProvider trustboundary.DataProvider
```

Comment:
This struct is exposing an internally defined struct. This struct would need to be defined elsewhere.

Reply:
To address this, I've introduced a public auth.TrustBoundaryDataProvider interface so the public API no longer exposes internal types. The internal providers are now wrapped in an adapter that implements this public interface.

For the data struct itself, auth.TrustBoundaryData is a public type alias, and its fields are now public for simplicity. This also addresses the previous immutability concerns by making it a simple data-carrying struct.